### DISCUSSION TITLE
Make "reopen closed tab" reopen last closed tree?

### DISCUSSION THREAD

**Discussion by TheOverpassArsonist** [piroor/treestyletab#3613] [Q&A]
Is there a way to make it so that when you click "reopen closed tab" it reopens all of the tabs in the most recently closed tree at once? 

TST seems to remember their ordering, but it won't reopen them all at once. (if I repeatedly "reopen closed tab" then they'll restore with the right tree structure, but they'll only do it one at a time)

**Comment by piroor** [comment]
TST actually tries to reopen a set of closed tabs if possible with the option: TST options => `Unlock Expert Options` and `Advanced` => `Restore the set of tabs most recently closed, when one of them is restored via the "Reopen Closed Tab"`. It is enabled by default but can be disabled accidentally.

However the feature sometimes fails to restore tree. Firefox does not provide any API to detect tab restoration, and TST guesses a tab is restored or not based on some peripheral information, thus the detection of "one of restored tabs" is fragile.
